### PR TITLE
[WIP] Remove skip from compute_build_matrix

### PR DESCRIPTION
**Pull request by johanneskoester**
Instead, fix issue #488 by changes to conda-build-all, see SciTools/conda-build-all#82.

Currently, this PR pulls conda-build-all from my fork for demonstration. Once SciTools/conda-build-all#82 has been merged and released, we can update the dependency of conda-smithy.

### CODE REVIEW

**Comment by ocefpaf** [comment]
Thanks @johanneskoester this is much better then #488 indeed!

File: .travis.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ install:
     # Now do the things we need to do to install it.
     - conda install --file requirements.txt coverage=3 python-coveralls mock python=${PYTHON} ${CONDA_PKGS} --yes --quiet -c conda-forge
     - python setup.py install
+    - pip install git+https://github.com/johanneskoester/conda-build-all.git
```
Comment (ocefpaf) [inline .travis.yml]: If SciTools folks takes too long to merge that we can add this as a patch on our own packaged version of conda-build-all and drop the patch once they merge it.

**Comment by jakirkham** [comment]
Maybe cleanup the commit history a bit. Good once we have a `conda-build-all` package ready.

**Comment by johanneskoester** [comment]
Don't you do a squash or merge commit?

**Comment by johanneskoester** [comment]
Closing in favor of #493.

**Pull request closed without merging**